Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
+  - set `contract_timeout_cltv` sufficiently long after `contract_maturity_cltv` to allow for late oracle signatures and other delays to closing the contract.
 
-### Accept
-After receiving the `offer` message from A, and after [validating](#offer-validation) the provided information, B creates and sends back an `accept` message containing:
-1. B's fund amount (satoshi)
-1. B's public keys
-1. B's addresses
-1. B's inputs
-1. CET signatures
-1. Refund signature
+The receiving node MUST:
 
-#### B's fund amount (satoshi)
-How much B inputs into the contract.
+  - ignore undefined bits in `channel_flags`.
 
-#### B's public keys
-The following public keys:
+The receiving node MAY reject the contract if:
 
-1. Funding Public Key
-   - B's public key used in the multi-signature [funding output](Transactions.md#FundingOutputs)
-2. Sweep Public Key
-   - B's public key used in the creation of the CETs [output script](Transactions.md#CETOutputs).
+  - it does not agree to the terms in `contract_info`.
+  - the `contract_info` is missing relevant events.
+  - it does not want to use the oracle(s) specified in `oracle_info`.
+  - `total_collateral_satoshis` is too small.
 
-#### B's addresses
+The receiving node MUST reject the contract if:
 
-The following addresses:
+  - the `chain_hash` value is set to a hash of a chain that is unknown to the receiver.
+  - the `contract_info` refers to events unknown to the receiver.
+  - the `oracle_info` refers to an oracle unknown to the receiver.
+  - it considers `feerate_per_vb` too small for timely processing or unreasonably large.
+  - `funding_pubkey` is not a valid secp256k1 pubkey in compressed format.
+  - the funder's amount for the funding transaction is not sufficient for full [fee payment](Transactions.md#fee-payment).
 
-1. Funding Change Address
-   - The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
-2. Final Address
-   - The address to which funds will be sent in [unilateral contract execution](Transactions.md#CETOutputs), [refund](Transactions.md#refund-transaction) and [mutual closing](Transactions.md#mutual-closing-transaction) transactions.
+### The `accept_dlc` Message
 
-#### B's inputs
-The set of UTXOs to be used as input to the fund transaction.
+This message contains information about a node and indicates its
+acceptance of the new DLC, as well as its CET and refund transaction
+signatures. This is the second step toward creating the funding transaction
+and closing transactions.
 
-#### CET signatures
-A set of signatures from B, one for each CET input.
+1. type: ??? (`accept_dlc`)
+2. data:
+   * [`32*byte`:`temporary_contract_id`]
+   * [`u64`:`total_collateral_satoshis`]
+   * [`point`:`funding_pubkey`]
+   * [`spk`:`payout_spk`]
+   * [`u16`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`cet_signatures`:`cet_signatures`]
```

Comment:
again I feel here the type should be `nb_cets*signature`.

Reply:
But we need a label on each signature to point it to what outcome it signs for no? Or I guess we could use ordering explicitly (like copying the structure of the contract info) but this would need to be added to the specification if this were the case. I'll probably hold off judgement until the PR where `cet_signatures` as a TLV is defined.

Reply:
Yes I was assuming that the order of signatures and outcomes was corresponding, but I feel this is an acceptable assumption, as I can't think of a reason not to do that.

Reply:
It would be nice to decide something on this. One reason I can think of to follow your approach is that we might have to split them in several messages in which case having the index can be practical.

Reply:
https://github.com/discreetlogcontracts/dlcspecs/issues/76